__tests__/app.test.js

```diff
@@ -37,290 +37,322 @@ describe("Northcoders News API ", () => {
+          });
+      });
+      test("Should send code 404 and message if article does not exist", () => {
+        return request(app)
+          .get("/api/articles/999/comments")
+          .expect(404)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe(`article_id 999 does not exist`);
+          });
+      });
+      describe("POST /api/articles/:article_id/comments", () => {
+        const commentThis = {
+          username: "butter_bridge",
+          body: `Did you just finish all the coffee?!`,
+        };
+        const matchedShape = {
+          comment_id: 19,
+          body: `Did you just finish all the coffee?!`,
+          article_id: 3,
+          author: "butter_bridge",
+          votes: 0,
+        };
+        test("Should return status 201 when new comment successfully posted", () => {
+          return request(app)
+            .post("/api/articles/3/comments")
+            .send({ ...commentThis })
+            .then(() => {
+              expect(201);
+            });
+        });
+        test("Should return posted comment showing all desired keys", () => {
+          return request(app)
+            .post("/api/articles/3/comments")
+            .send({ ...commentThis })
+            .then(({ body }) => {
+              expect(body.comment).toMatchObject({ ...matchedShape });
+            });
+        });
+        test("Should return 400 and error message if user attempts to post empty body", () => {
+          return request(app)
+            .post("/api/articles/1/comments")
+            .send({ username: "Monstera_Munch" })
+            .then(({ body: { msg } }) => {
+              expect(400);
+              expect(msg).toBe(`Nothing to post!`);
+            });
+        });
+        test("Should return 400 and error message if user attempts to post anonymously", () => {
+          return request(app)
+            .post("/api/articles/1/comments")
+            .send({ body: "The mystery poster strikes again!" })
+            .then(({ body: { msg } }) => {
+              expect(400);
+              expect(msg).toBe(`Malformed request body, missing username key`);
+            });
+        });
+      });
+    });
   });
-});
\ No newline at end of file
+
+
+  describe("GET /api/users", () => {
+    test("Should return status 200", () => {
+      return request(app)
+        .get("/api/users")
+        .expect(200);   
+    })
+    test("Should return an object with all users in array on a key of 'users'", () => {
+      return request(app)
+        .get("/api/users")
+        .then(({ body: { users } }) => {
+          expect(users.length).toBe(4)
+          users.forEach((user) => {
+            expect(user).toHaveProperty("username");
+            expect(user).toHaveProperty("name");
+            expect(user).toHaveProperty("avatar_url");
+          });
+        });
+    });
+    test("Each object in the users array should only have 4 keys", () => {
```

Comment:
I wouldn't include this test. Imagine if you wanted to add extra properties to a user. Then this would break.